app/src/main/java/com/example/she/DatabaseHandler.java

```diff
@@ -32,22 +33,41 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
     }
 
     //to add new contact
-    public boolean insertData(String name , String num){
+    public boolean insertData(String name, String num) {
         SQLiteDatabase db = this.getWritableDatabase();
         ContentValues contentValues = new ContentValues();
-        contentValues.put(COL_2,name);
-        contentValues.put(COL_3,num);
-        long result = db.insert(TABLE_NAME,null,contentValues);
-        if(result == -1)
+        contentValues.put(COL_2, name);
+        contentValues.put(COL_3, num);
+        long result = db.insert(TABLE_NAME, null, contentValues);
+        if (result == -1)
             return false;
         else
             return true;
     }
 
     //to get all contacts
-    public Cursor getAllContacts(){
+    public Cursor getAllContacts() {
         SQLiteDatabase db = this.getWritableDatabase();
-        Cursor cursor = db.rawQuery("SELECT * FROM " + TABLE_NAME,null);
+        Cursor cursor = db.rawQuery("SELECT * FROM " + TABLE_NAME, null);
         return cursor;
     }
+
+    //to delete selected contact
+    public boolean deleteContact(String phoneNumber) {
+
+        SQLiteDatabase db = this.getWritableDatabase();
+
+        String deleteQuery = "DELETE FROM " + TABLE_NAME + " WHERE " + COL_3 + " = " + phoneNumber;
+        Cursor cursor = db.rawQuery(deleteQuery, null);
+
+        if (cursor.moveToFirst()) {
```

Comment:
Good Job, bro ! 
Just a minor updation:
In DatabaseHandler.java, remove lines 64,65,68 and 69 and add lines 64 and 65 outside if-else. It helps avoiding unnecessary code repetition.

Reply:
But then error will occur because in that case we are accessing cursor after closing the database and cursor.

Reply:
Oh Yeah ! You are right. Missed this point.
Although, You can write db.close() outside the if-else :D

Reply:
I have already tried that but then also it shows error.